pkg/loki/modules.go

```diff
@@ -975,8 +975,14 @@ func (t *Loki) setupAsyncStore() error {
 }
 
 func (t *Loki) initIngesterQuerier() (_ services.Service, err error) {
-	logger := log.With(util_log.Logger, "component", "querier")
-	t.ingesterQuerier, err = querier.NewIngesterQuerier(t.Cfg.IngesterClient, t.ring, t.Cfg.Querier.ExtraQueryDelay, t.Cfg.MetricsNamespace, logger)
+	logger := log.With(util_log.Logger, "component", "ingester-querier")
+
+	if !t.Cfg.Ingester.KafkaIngestion.Enabled {
+		// Disable querying partition-ingesters if Kafka ingestion is disabled
+		t.Cfg.Querier.QueryPartitionIngesters = false
```

Comment:
Is this "fixing" invalid configuration (i.e. the case where someone has querying partition ingesters enabled but kafka ingestion disabled at the same)?

Reply:
Yes - do you think this should be a validation check instead? I thought the intended effect & the solution were clear so I could just fix it on-the-fly, but, looking again, I think a validation would make more sense to avoid confusing the user. I'll add it!

Reply:
Actually, this has the same problem as the other config params: They can be used in various modes (SSD, All-in-one, microservices) and enabled via per-component flags & config files. A validation would only help in the All-in-one case so I'm going to remove this check instead.